scripts/install/installer

```diff
@@ -974,12 +985,16 @@ def upgrade_cluster(ctx, development):
     config = ctx.obj
     cluster = Cluster(config, development)
     cluster.upgrade()
-    pnames = get_pnodes_names(config)
-    pnode_action(pnames, 'stop', config)
-    pnode_action(pnames, 'update', config)
-    cluster.update()
-    pnode_action(pnames, 'start', config)
-    print('Updating cluster done.')
+    try:
+        pnames = get_pnodes_names(config)
+        pnode_action(pnames, 'stop', config)
+        pnode_action(pnames, 'update', config)
+        cluster.update()
+        pnode_action(pnames, 'start', config)
+        print('Updating env done.')
+    except:
+        cluster.revert()
+        print("Failed to update env.")
```

Comment:
will we have logs what exactly went wrong at this point?

Reply:
We will have the logs up until this moment and then this message.